waku/v2/node/jsonrpc/relay_api.nim

```diff
@@ -1,12 +1,32 @@
+{.push raises: [Exception, Defect].}
+
 import
+  std/[tables,sequtils],
   json_rpc/rpcserver,
+  libp2p/protocols/pubsub/pubsub,
   eth/[common, rlp, keys, p2p],
   ../../waku_types,  
   ../wakunode2,
   ./jsonrpc_types, ./jsonrpc_utils
 
+const futTimeout = 5.seconds
+
 proc installRelayApiHandlers*(node: WakuNode, rpcsrv: RpcServer) =
-  const futTimeout = 5.seconds
+  ## Create a per-topic message cache
+  var
+    topicCache = initTable[string, seq[WakuMessage]]()
```

Comment:
Hm, does this state belong in JSONRPC code? I suppose that's the only place it is currently used. Seems a bit odd to me though, with this local state? Not sure.

Reply:
Good question. After considering a couple of options I settled on this being the _safest_ place for the state, where it is as short- or long-lived as the RPC handlers and will be garbage collected with everything else. Still, I will definitely consider a better (more global) place for this, though asynchronous read/write access to the table may become tricky.